### ISSUE TITLE
How to generate retrieved_trials.json

### ISSUE THREAD

**Issue by traycho** [ncbi-nlp/TrialGPT#1]
I could not find a script that generates  - retrieved_trials.json 

Imagine I have a brand new  dataset/new and have underlying 
```
  /qrels/tests.tsv - empty
  /corpus.jsonl - populates as per other corpuses
  /id2queries.json - a single patient 
  /queries.jsonl - a single record for this patient 
 ``` 
  
  I would like to generate a file retrieved_trials.json for this new corpus.

**Comment by drqiaojin** [comment]
Can you run https://github.com/ncbi-nlp/TrialGPT?tab=readme-ov-file#trialgpt-retrieval?

**Comment by traycho** [comment]
Yes, I have run generated keywords but it generates a file  `/results/retrieval_keywords_XXXX_YYY.json` the `retrieved_trials.json` is empty for me. 

Where in the code you do populate this file ?

**Comment by drqiaojin** [comment]
> Yes, I have run generated keywords but it generates a file `/results/retrieval_keywords_XXXX_YYY.json` the `retrieved_trials.json` is empty for me.
> 
> Where in the code you do populate this file ?

The information is essentially generated here: `https://github.com/ncbi-nlp/TrialGPT/blob/main/trialgpt_retrieval/hybrid_fusion_retrieval.py#L224`, but in the `retrieved_trials.json` we further organized the retrieved trials by their relevance judgments. That processing code is not available as for now but you are welcome to make a PR for that.

Thanks,
Qiao

**Comment by Deepadsc** [comment]
HI @Andy-jqa 

I am able to run hybridfusion.py, but after running it, I got the final results with only NCTIDs as per the code. However, I tried to reorganize the structure to match the retrieved_Trials.json format.

There is one thing I am not understanding—I can retrieve the metadata of the relevant NCTIDs for each patient, but I see three categories: "0", "1", and "2". On what basis are these three categories defined?

I used qrels with values 0, 1, and 2, but they are not matching. Can you help me find a solution for this?

**Comment by aaron-zhlg** [comment]
I noticed that for each patient, the trials listed in retrieved_trials.json seem to exactly match the relevant trials in test.tsv, with no missing ones. Since retrieved_trials.json is described as being generated by GPT-4-based retrieval, I’m a bit confused — GPT-4 retrieval shouldn’t have perfect recall, right?
Just wondering if this file was post-processed to ensure full coverage of the ground-truth trials?
